src/services/connections.js

```diff
@@ -219,4 +266,77 @@ module.exports = class ConnectionHelper {
 			throw error
 		}
 	}
+
+	/**
+	 * Fetch a list of connections based on query parameters and filters.
+	 * @param {number} pageNo - The page number for pagination.
+	 * @param {number} pageSize - The number of records per page.
+	 * @param {string} searchText - The search text to filter results.
+	 * @param {Object} queryParams - The query parameters for filtering.
+	 * @param {string} userId - The ID of the authenticated user.
+	 * @param {string} orgId - The organization ID for filtering.
+	 * @returns {Promise<Object>} A list of filtered connections.
+	 */
+	static async list(pageNo, pageSize, searchText, queryParams, userId, orgId) {
+		try {
+			let organizationIds = []
+
+			if (queryParams.organization_ids) {
+				organizationIds = queryParams.organization_ids.split(',')
+			}
+
+			const query = utils.processQueryParametersWithExclusions(queryParams)
+			const userExtensionsModelName = await menteeQueries.getModelName()
+
+			// Fetch validation data for filtering connections
+			const validationData = await entityTypeQueries.findAllEntityTypesAndEntities({
```

Comment:
@nevil-mathew Please create a story for 3.2 which will allow to configure which entity type to show in the listing and which to show in detail page. Also remove designation, pass everything here.